mmcv/ops/nms.py

```diff
@@ -14,21 +14,24 @@
 class NMSop(torch.autograd.Function):
 
     @staticmethod
-    def forward(ctx, bboxes, scores, iou_threshold, score_threshold, max_num,
-                offset):
-        valid_mask = scores > score_threshold
-        bboxes, scores = bboxes[valid_mask], scores[valid_mask]
-        valid_inds = torch.nonzero(valid_mask, as_tuple=False).squeeze(dim=1)
+    def forward(ctx, bboxes, scores, iou_threshold, offset, score_threshold,
+                max_num):
+        if torch.onnx.is_in_onnx_export():
+            valid_mask = scores > score_threshold
+            bboxes, scores = bboxes[valid_mask], scores[valid_mask]
+            valid_inds = torch.nonzero(
```

Comment:
NonZero op is dynamic and should be avoided because we have to support TensorRT as well, which do not have NonZero op.

Reply:
If TensorRT works with ONNX graph and supports NMS operation, this shouldn't be a problem.

When the `symbolic` function is implemented, it replaces the `forward` function in the ONNX graph.
So there should be no problem with dynamic NonZero.

This is how the current NMS operation looks in the ONNX graph:
<img width="416" alt="nms" src="https://user-images.githubusercontent.com/10177297/115555708-7b39cd80-a2b8-11eb-94ac-cb23f901bbd6.PNG">

Reply:
In that case, it's OK.